src/main/webapp/app/course/manage/overview/course-management-exercise-row.component.ts

```diff
@@ -0,0 +1,96 @@
+import { ChangeDetectionStrategy, Component, Input, OnChanges } from '@angular/core';
+import { Exercise, ExerciseType } from 'app/entities/exercise.model';
+import { IconProp } from '@fortawesome/fontawesome-svg-core';
+import { CourseManagementOverviewExerciseStatisticsDTO } from 'app/course/manage/overview/course-management-overview-exercise-statistics-dto.model';
+import { Course } from 'app/entities/course.model';
+import { round } from 'app/shared/util/utils';
+
+export enum ExerciseRowType {
+    FUTURE = 'future',
+    CURRENT = 'current',
+    ASSESSING = 'assessment',
+    PAST = 'past',
+}
+
+@Component({
+    selector: 'jhi-course-management-exercise-row',
+    templateUrl: './course-management-exercise-row.component.html',
+    styleUrls: ['course-management-exercise-row.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class CourseManagementExerciseRowComponent implements OnChanges {
+    @Input() course: Course;
+    @Input() details: Exercise;
+    @Input() statistic: CourseManagementOverviewExerciseStatisticsDTO;
+    @Input() rowType: ExerciseRowType;
+
+    // Expose enums to the template
+    exerciseType = ExerciseType;
+    exerciseRowType = ExerciseRowType;
+    quizStatus = {
+        CLOSED: 'CLOSED',
+        OPEN_FOR_PRACTICE: 'OPEN_FOR_PRACTICE',
+        ACTIVE: 'ACTIVE',
+        VISIBLE: 'VISIBLE',
+        HIDDEN: 'HIDDEN',
+    };
+
+    JSON = JSON;
+
+    hasLeftoverAssessments = false;
+    averageScoreNumerator: number;
+    icon: IconProp;
+    iconTooltip: string;
+
+    getIcon(type: ExerciseType | undefined): IconProp {
```

Comment:
are you sure that this has not been implemented somewhere else? at least we already use this somewhere else